game/world/managers/objects/spell/SpellEffectHandler.py

```diff
@@ -37,10 +37,15 @@ def apply_effect(casting_spell, effect, caster, target):
                          f'{effect.effect_type}) from spell {casting_spell.spell_entry.ID}.')
             return
 
+        allowed_effects_on_death = {
+            SpellEffects.SPELL_EFFECT_RESURRECT,
+            SpellEffects.SPELL_EFFECT_SUMMON_OBJECT_WILD  # Allow spells to summon an object on creature death.
+        }
+
         from game.world.managers.objects.units.UnitManager import UnitManager
         if target and isinstance(target, UnitManager):
-            # Do not apply spell effects on dead targets unless it's a resurrection effect.
-            if not target.is_alive and effect.effect_type != SpellEffects.SPELL_EFFECT_RESURRECT:
+            # Do not apply spell effects on dead targets unless it's an allowed effect.
+            if not target.is_alive and effect.effect_type not in allowed_effects_on_death:
                 return
```

Comment:
I don't think this is a proper fix for the issue. This behavior isn't related to the effect, but rather the AI on-death context. It will still fail for other on-death effects. I'd suggest moving the AI death trigger to run just before death instead.

Reply:
I implemented it that way to prevent the race condition that the mob is already dead before the server casts the spell. 

I just tried to do it the way you suggested and it didn't work. When I moved the AI death trigger to line 619 of CreatureManager just before 

```        was_oneshot = not self.threat_manager.has_aggro_from(killer)```

the trigger was called but the spell was never cast. And that's at the earliest possible point I could find where the creature wasn't already flagged as dead.

There are 189 AI events which are triggered on death, many of which aren't relevant to us because they are used by creatures not used in 0.5.3 or run scripts that aren't casting spells. There are a few more occurrences of a spellcast on death - I actually missed an important one but added it right now - but I am certain I can find out which ones are needed and add them.